app/models/measure_condition.rb

```diff
@@ -132,4 +134,16 @@ def universal_waiver_applies?
   def units
     measure_condition_components.map(&:unit)
   end
+
+  def permutation_key
+    return unless certificate_type_code || certificate_code || condition_duty_amount
```

Comment:
So AFAIK the condition either has a certificate type code and certificate code _or_ a condition duty amount.

Reply:
The actual condition means that all `certificate_type_code, certificate_code, condition_duty_amount` must be true to return 
"#{certificate_type_code}-#{certificate_code}-#{condition_duty_amount}"
but @willfish is suggesting that  the condition should be
`if (certificate_type_code && certificate_code)  ||  condition_duty_amount`

I don't know, just thinking out loud about this 😄

Reply:
@willfish yeh - I'd thought that rule should be `(certificate_type_code && certificate_code) || condition_duty_amount)` but this was changed at Matts request so I think its right.

@alexdesi the condition is _any_ of them, not _all_ of them - its just used for the grouping so if it ends up some like "AB--" as a permutation key thats fine - this isn't publicly exposed, its just an implementation detail